Sources/Diagnose/DiagnoseCommand.swift

```diff
@@ -222,7 +223,12 @@ public struct DiagnoseCommand: AsyncParsableCommand {
             // The log is larger than we expected. Halt at 100%
             progress = 1
           }
-          reportProgress(.collectingLogMessages(progress: progress), message: "Collecting log messages")
+          if progress - lastReportedProgress > 0.01 {
+            // Only call `reportProgress` when we've made at least 1% of progress (which will get scaled down further
+            // based on the other components). This avoids us spamming the console with non-meaningful progress updates.
```

Comment:
IMO it seems like this is something that should be handled by `reportProgress`, but I don't feel strongly about it

Reply:
Yes, that’s probably the better design. Updated.